src/lib.rs

```diff
@@ -191,36 +262,37 @@ impl<'entry, 'b> Iterator for Structures<'entry, 'b> {
         self.idx = strings_idx + strings_len;
         self.count += 1;
 
-        Some(Ok(structure))
+        Some(match structure.info {
+            InfoType::System => System::new(structure).map(Structure::System),
+            InfoType::BaseBoard => BaseBoard::new(structure).map(Structure::BaseBoard),
+            InfoType::Processor => Processor::new(structure).map(Structure::Processor),
+            _ => Ok(Structure::Other(structure))
+        })
     }
 }
 
+#[doc(hidden)]
 #[repr(C)]
 #[repr(packed)]
-pub struct HeaderPacked {
+struct HeaderPacked {
     pub _type: u8,
```

Comment:
Why `pub` but with the '_'?

Reply:
Aha! Oh yeah, the `_type` is because you cannot name something `type` because that is a reserved word.